hr_expense_product_policy/models/hr_expense.py

```diff
@@ -4,6 +4,7 @@
 
 from openerp import api, fields, models
 from openerp.tools import ustr
+from openerp.exceptions import Warning as UserWarning
```

Comment:
AFAIK `Warning`is deprecated. Use `ValidationError` or  `UserError` instead.
I believe that you can avoid an extra import and use `raise models.ValidationError`